llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -2985,6 +3094,23 @@ void InnerLoopVectorizer::fixVectorizedLoop(VPTransformState &State) {
   PSE.getSE()->forgetLoop(OrigLoop);
   PSE.getSE()->forgetBlockAndLoopDispositions();
 
+  // When dealing with uncountable early exits we create a vector.early.exit
+  // block predecessor of the exit block. This block needs to have the same
+  // parent loop as the exit block itself.
+  VPRegionBlock *VectorRegion = State.Plan->getVectorLoopRegion();
+  if (BasicBlock *OrigEarlyExitBB = Legal->getUncountableEarlyExitBlock()) {
```

Comment:
I had to fix a bug in `VPBasicBlock::execute` that incorrectly adds some VPBasicBlocks to the wrong parent loop.